### PR TITLE
(BIDS-2178) the next is justified

### PR DESCRIPTION
**Pull request by MauserBitfly**
### <samp>🤖 Generated by Copilot at d6b62d6</samp>

This pull request adds a block status badge to the slot overview page, which indicates if the block is finalized or not, and if the previous epoch was finalized or not. It also enhances the slot visualization page with more information and functionality. It modifies the `handlers/slot.go`, `templates/slotViz.html`, `templates/slot/overview.html`, and `types/templates.go` files.

### CODE REVIEW

File: handlers/slot.go [hunk 1]
```diff
@@ -245,13 +246,15 @@ func GetSlotPageData(blockSlot uint64) (*types.BlockPageData, error) {
 		LEFT JOIN blocks_tags ON blocks.slot = blocks_tags.slot and blocks.blockroot = blocks_tags.blockroot
 		LEFT JOIN tags ON blocks_tags.tag_id = tags.id
 		LEFT JOIN epochs ON (GREATEST(blocks.slot,1)-1)/$2 = epochs.epoch
+		LEFT JOIN epochs prev_epoch ON GREATEST(((GREATEST(blocks.slot,1)-1)/$2-1),0) = prev_epoch.epoch
```
Comment (LuccaBitfly) [inline handlers/slot.go:249]:
```suggestion
		LEFT JOIN epochs prev_epoch ON GREATEST((blocks.slot-1)/$2-1,0) = prev_epoch.epoch
```
Doesn't this do the same thing?

**Review by LuccaBitfly — CHANGES_REQUESTED** [review]
Works fine, one small issue

File: handlers/slot.go (hunk 1 above)
Comment (MauserBitfly) [reply handlers/slot.go:249]: No that would not work for slot 0

Comment (MauserBitfly) [reply handlers/slot.go]: adapted your suggestion

**Review by recy21 — APPROVED** [review]
lgtm

**Pull request merged**